src/components/BarrierFree/BarrierFree.js

```diff
@@ -0,0 +1,177 @@
+      .reverse()
+      .filter(l => !l.getIsBaseLayer() && l.getVisible());
+
+    const mapExtent = map.getView().calculateExtent(map.getSize());
+
+    let visFeats = [];
+    visibleLayers.forEach(l => {
+      if (l && l.olLayer) {
+        const features = l.olLayer.getSource().getFeatures();
+        if (features.length > 0) {
+          features.forEach(f => f.set('layerRef', l));
+          const visibleFeats = features.filter(
+            feat =>
+              containsExtent(mapExtent, feat.getGeometry().getExtent()) &&
+              filterFeat(feat),
+          );
+          // Check if visible on the map
+          visFeats = visFeats.concat(visibleFeats);
+        }
+      }
+    });
+    visFeats.sort((a, b) => {
+      if (
+        a.getGeometry().getCoordinates()[0] <
+        b.getGeometry().getCoordinates()[0]
+      ) {
+        return -1;
+      }
+      return 1;
+    });
+    visibleFeatures = visFeats;
+  };
+
+  const focusFeature = () => {
+    const featToFocus = visibleFeatures[featureIndex];
+    featLayer = featToFocus ? featToFocus.get('layerRef') : null;
+    if (featLayer) {
+      tabFeature = true;
+      featLayer.clickedFeature = featToFocus;
+      featLayer.olLayer.changed();
+    }
+  };
+
+  const onFocus = () => {
+    setVisibleFeats();
+    tabFeature = true;
+    featureIndex = featureIndex || -1;
+    focusFeature();
+  };
+
+  const onKeyUp = e => {
+    const forward = !e.shiftKey;
+    if (!tabFeature) {
+      document.body.classList.add(focusClass);
+    } else {
+      featureIndex += forward ? 1 : -1;
+
+      if (forward && featureIndex >= visibleFeatures.length) {
+        removeHighlight();
+      } else if (!forward && featureIndex <= -1) {
+        removeHighlight();
+        featureIndex = -1;
+      } else {
+        focusFeature();
+      }
+      e.preventDefault();
+      e.stopPropagation();
+    }
+  };
+
+  const onDocumentClick = () => {
+    // Remov bf class from the document body.
+    document.body.classList.remove(focusClass);
+    // Reset value to default.
+    tabFeature = false;
+  };
+
+  const onDocumentKeyUp = e => {
+    if (e.keyCode === 9) {
+      onKeyUp(e);
+    }
+    if ([38, 40].includes(e.keyCode)) {
+      onKeyUp(e, document.activeElement);
+    }
+  };
+
+  const onDocumentKeyDown = e => {
+    if (e.keyCode === 9) {
+      const forward = !e.shiftKey;
+      if (
+        !tabFeature ||
+        (forward && featureIndex >= visibleFeatures.length) ||
+        (!forward && featureIndex <= -1)
+      ) {
+        return;
+      }
+      e.preventDefault();
+      e.stopPropagation();
+    }
+  };
+
+  useEffect(() => {
+    setVisibleFeats();
+  }, [center, zoom, resolution, layerService, setVisibleFeats]);
+
+  useEffect(() => {
+    // ComponentDidMount
+    document.addEventListener('click', onDocumentClick);
+    document.addEventListener('keyup', onDocumentKeyUp);
+    document.addEventListener('keydown', onDocumentKeyDown);
```

Comment:
I think the functions  onDocumentXXX  must be defined in the `useEffect`. Have you verify if the event is properly remove ?